polar-core/src/warnings.rs

```diff
@@ -204,3 +204,53 @@ pub fn check_ambiguous_precedence(rule: &Rule, kb: &KnowledgeBase) -> PolarResul
     walk_rule(&mut visitor, rule);
     visitor.warnings()
 }
+
+pub fn check_no_allow_rule(kb: &KnowledgeBase) -> PolarResult<Vec<String>> {
+    let has_allow_rule = kb.get_rules().iter().any(|(name, _)| name.0 == "allow");
+    if has_allow_rule {
+        Ok(vec![])
+    } else {
+        Ok(vec![
+            "Your policy does not contain an allow rule, which means \
+that no actions are allowed. Did you mean to add an allow rule to \
```

Comment:
What if someone intentionally isn't using `is_allowed/3` or `authorize/3` in their policy?

Case 1: using `query_rule()` to query a non-`allow` entrypoint rule.

Case 2: using `authorize_request/2` instead of `authorize/3`, so `allow_request/2` is the entrypoint rule instead of `allow/3`. Same thing for folks authorizing fields instead of requests.

Reply:
Hmm yeah this is a good point. I don't think it's the end of the world to have a bit of a noisy warning in that case, but it would be great if users could tell us to ignore specific warnings (see my other comment). I think right now making this warning ignore-able would introduce more complexity than its worth (probably would have to use a random environment variable). 

To make this warning more "semantically correct", perhaps I could add a "usually" to the warning message?
```suggestion
            "Your policy does not contain an allow rule, which usually means \
that no actions are allowed. Did you mean to add an allow rule to \
```

Reply:
What do you think about updating the check to look for `allow` or `allow_request` or `allow_field`?

Reply:
Good idea, updated

Reply:
Note that I still made the warning message primarily about `allow` -- I feel like that's the common thing to point people towards, but it still mentions `allow_request` and `allow_field` for correctness